### PR TITLE
Update PipelineRun and PipelineRuns when namespace changes

### PR DESCRIPTION
**Pull request by AlanGreene**
# Changes

Dispatch appropriate actions when namespace changes and either
PipelineRun or PipelineRuns are mounted. This will ensure
the APIs are called to get fresh data for the selected namespace.

https://github.com/tektoncd/dashboard/issues/40
# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [ ] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [ ] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/dashboard/blob/master/CONTRIBUTING.md)
for more details._

### CODE REVIEW

**Comment by AlanGreene** [comment]
@steveodonovan this doesn't address URL changes for these 2 components which needs to be handled in a slightly different way and will be in a separate PR.

File: src/containers/PipelineRuns/PipelineRuns.js [hunk 1]
```diff
@@ -30,11 +30,23 @@ import { fetchPipelineRuns } from '../../actions/pipelineRuns';
...
 
 export /* istanbul ignore next */ class PipelineRuns extends Component {
-  async componentDidMount() {
+  componentDidMount() {
+    this.fetchPipelineRuns();
+  }
+
+  componentDidUpdate(prevProps) {
+    const { namespace } = this.props;
+    if (namespace !== prevProps.namespace) {
```
Comment (steveodonovan) [inline src/containers/PipelineRuns/PipelineRuns.js]: We also want to do this if pipelineName has changed

File: src/containers/PipelineRun/PipelineRun.js [hunk 2]
```diff
@@ -55,13 +56,14 @@ export /* istanbul ignore next */ class PipelineRunContainer extends Component {
...
-      this.props.fetchPipelineRun(pipelineRunName),
-      this.props.fetchTasks(),
-      this.props.fetchTaskRuns()
-    ]).then(() => this.setState({ loading: false }));
+    this.fetchData();
+  }
+
+  componentDidUpdate(prevProps) {
+    const { namespace } = this.props;
+    if (namespace !== prevProps.namespace) {
```
Comment (steveodonovan) [inline src/containers/PipelineRun/PipelineRun.js]: Ditto if pipeline run name has changed

File: src/containers/PipelineRuns/PipelineRuns.js (hunk 1 above)
Comment (AlanGreene) [reply src/containers/PipelineRuns/PipelineRuns.js]: Done 👍

File: src/containers/PipelineRun/PipelineRun.js (hunk 2 above)
Comment (AlanGreene) [reply src/containers/PipelineRun/PipelineRun.js]: Done 👍

**Comment by AlanGreene** [comment]
@steveodonovan thanks for the review. I've updated to address both of your comments. While testing this I discovered an existing issue with PipelineRun where we don't actually use the pipelineName segment of the URL, this means that segment could have any arbitrary value and is ignored. For completeness we should validate this and I've created an issue to track: https://github.com/tektoncd/dashboard/issues/142

**Pull request merged**